packages/eslint-plugin-react-hooks/__tests__/ESLintPureRender-test.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const ESLintTester = require('eslint').RuleTester;
+const ReactHooksESLintPlugin = require('eslint-plugin-react-hooks');
+const PureRenderESLintRule = ReactHooksESLintPlugin.rules['pure-render'];
+
+ESLintTester.setDefaultConfig({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+const writeError = {
+  message:
+    'Writing to refs during rendering is not allowed. Move this into a useEffect or useLayoutEffect. See https://beta.reactjs.org/apis/useref',
+};
+const readError = {
+  message:
+    'Reading from refs during rendering is not allowed. See https://beta.reactjs.org/apis/useref',
+};
```

Comment:
I would like some help with these error messages. I'm not sure exactly how much info we'd like to expose in the message vs in the docs link.

Also, would be nice if we could link directly to the "Pitfall" heading on that docs page, but right now it doesn't seem to have an anchor link.

Reply:
dmddj

Reply:
Good job.